src/main/java/org/openrewrite/java/testing/junit5/TimeoutRuleToClassAnnotation.java

```diff
@@ -0,0 +1,129 @@
+            if (javaParser == null) {
+                javaParser = JavaParser.fromJavaVersion()
+                        .classpathFromResources(ctx, "junit-jupiter-api-5", "hamcrest-3");
+            }
+            return javaParser;
+
+        }
+
+        @Override
+        public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+            J.ClassDeclaration cd = super.visitClassDeclaration(classDecl, ctx);
+
+            AtomicReference<Expression> initializer = new AtomicReference<>();
+
+            cd = cd.withBody(cd.getBody().withStatements(ListUtils.map(cd.getBody().getStatements(), statement -> {
+                if (statement instanceof J.VariableDeclarations) {
+                    //noinspection ConstantConditions
+                    if (TypeUtils.isOfClassType(((J.VariableDeclarations) statement).getTypeExpression().getType(),
+                            "org.junit.rules.Timeout")) {
+                        List<J.VariableDeclarations.NamedVariable> variables = ((J.VariableDeclarations) statement).getVariables();
+                        if (!variables.isEmpty()) {
+                            initializer.set(variables.get(0).getInitializer());
+                        }
+                        return null;
+                    }
+                }
+                return statement;
+            })));
+
+            if (initializer.get() != null) {
+                cd = insertTimeoutAnnotation(initializer.get(), cd, ctx);
+                maybeRemoveImport("org.junit.Rule");
+                maybeRemoveImport("org.junit.rules.Timeout");
+            }
+            return cd;
+        }
+
+        private J.ClassDeclaration insertTimeoutAnnotation(Expression ex, J.ClassDeclaration cd, ExecutionContext ctx) {
+            JavaTemplate.Builder builder;
+            Object[] params;
+            if (ex instanceof J.NewClass) {
+                List<Expression> arguments = ((J.NewClass) ex).getArguments();
+                if (arguments.size() == 2) {
+                    builder = JavaTemplate.builder("@Timeout(value = #{any(long)}, unit = #{any(TimeUnit)})");
+                    params = new Object[]{arguments.get(0), arguments.get(1)};
+                } else {
+                    builder = JavaTemplate.builder("@Timeout(value = #{any(long)}, unit = TimeUnit.MILLISECONDS)");
+                    params = new Object[]{arguments.get(0)};
+                }
+            } else if (ex instanceof J.MethodInvocation) {
+                String simpleName = ((J.MethodInvocation) ex).getName().getSimpleName();
+                String units = simpleName.equals("millis") ? "MILLISECONDS" : "SECONDS";
+                builder = JavaTemplate.builder("@Timeout(value = #{any(long)}, unit = TimeUnit." + units + ")");
+                params = new Object[]{((J.MethodInvocation) ex).getArguments().get(0)};
+            } else {
+                return cd;
```

Comment:
In this case we've already removed the rule above right? So for any unhandled case we'd be removing the rule without a  replacement. That to me indicates we should have similar restrictions when removing the rule as we do for adding the annotation.